tsl/src/bgw_policy/policy_utils.c

```diff
@@ -79,28 +79,46 @@ subtract_integer_from_now(int64 interval, Oid time_dim_type, Oid now_func)
 		case INT2OID:
 			res = DatumGetInt16(now) - interval;
 			if (res < PG_INT16_MIN || res > PG_INT16_MAX)
-				ereport(ERROR,
-						(errcode(ERRCODE_INTERVAL_FIELD_OVERFLOW), errmsg("ts_interval overflow")));
-			return Int16GetDatum(res);
+				*overflow = true;
```

Comment:
probably makes sense to use pg_sub_s16_overflow and pg_sub_s32_overflow for int2/int4

Reply:
Is there a reason why it wasn't implemented that way? Do you know? (This is refactored code from job.c)

Reply:
checked the pg_sub_s16_overflow and other variants. 
Makes this code a bit unwieldy to use the pg functions directly:  the result type  would  have to be int16, int32 , int64 for  sub_s16/sub_s32/sub_s64 respectively.  For example:
`       case INT2OID:
        {
            int16 res2 = 0;
            *overflow = pg_sub_s16_overflow(DatumGetInt16(now), interval, &res2);
            res = res2;
            break;
        }
`
I don't see a benefit in using pg_sub_sxx_overflow function for all the types since it is safe to do the subtraction using int64 types inline and just checking the range here.